src/murmur/Messages.cpp

```diff
@@ -155,6 +155,61 @@ bool isChannelEnterRestricted(Channel *c) {
 	return false;
 }
 
+/// Process the Capabilities message, used to negotiate voice protocol.
+/// See the doc in Mumble.proto for details about this message.
+///
+/// @param uSource ServerUser that receives this message.
+/// @param msg The message itself.
+void Server::msgCapabilities(ServerUser *uSource, MumbleProto::Capabilities &msg) {
+	MSG_SETUP(ServerUser::Connected);
+	int custom_index = 0;
+	for (int i = 0; i < msg.supported_protocols_size(); ++i){
+		auto protocol_type = msg.supported_protocols(i);
+		switch (protocol_type) {
+			case MumbleProto::Capabilities_VoiceProtocol_UDP_OCB2:
+				uSource->qlSupportedVoiceProtocols.append(VoiceProtocol(VoiceProtocolType::UDP_OCB2));
+				break;
+			case MumbleProto::Capabilities_VoiceProtocol_CUSTOM:
+				uSource->qlSupportedVoiceProtocols.append(VoiceProtocol(msg.custom_protocols(custom_index)));
+				++custom_index;
+				break;
+			default:
+				continue;
+		}
+	}
+	uSource->vptVoiceProtocolType = VoiceProtocolType::UNDEFINED;
```

Comment:
This should be the default value anyways. Why is it explicitly set here? :thinking:

Reply:
Because this message can also be used to re-negotiate the protocol.

Reply:
Got it :+1: